### DISCUSSION TITLE
JavaScript Starter Project for `xdeployer`

### DISCUSSION THREAD

**Discussion by defido** [pcaversaccio/xdeployer#20] [Show and tell]
[Firstbase JS single contract starter](https://github.com/defido/firstbase2) - Currently in pre beta but working. Don't recommend using in production, yet. 

After being asked if there was a JS starter, I have begun working on a JS specific starter. 

I think it's good that xDeployer is JS/TS agnostic and that projects can pick or choose whichever to use. Defido will continue to develop ideas & inputs into the xDeployer plugin + your typescript starter. Such as the refactoring of the ENV we did before. If we/I have a good idea on the JS starter, we/I will bring that idea to the xDeployer & see if that is something you would like to put into your TS starter. 

WHY:

There are certain projects that will prefer to use JS instead of TS files.

WHAT IS IT:

A simple starter JS project that uses a standard JS setup to complement the TS starter.

THIS WILL:

Be a JS specific starter with certain contracts such as OZ starter, reflections, taxes starters etc
Make projects choose their own testing tool suite.
Developed by the Defido side.

THIS WILL NOT:

Deviate dramatically from the typescript starter.

Great ideas will still be shared, and xDeployer will still be the focus :D

Thanks!

**Comment by pcaversaccio** [comment]
In order to put some context here, @defido is referring to my fully-fledged Hardhat project template [here](https://github.com/pcaversaccio/hardhat-project-template-ts) which is based on TypeScript.
